pysoa/common/transport/redis_gateway/backend/sentinel.py

```diff
@@ -0,0 +1,124 @@
+from __future__ import absolute_import, unicode_literals
+
+import itertools
+import random
+import time
+
+import redis
+import redis.sentinel
+import six
+
+from pysoa.common.transport.redis_gateway.backend.base import BaseRedisClient
+
+
+class SentinelRedisClient(BaseRedisClient):
+    """
+    Variant of the Redis client that supports the Redis Sentinel HA protocol.
+
+    Supports sharding, but assumes that there is only one sentinel cluster with multiple Redis services
+    monitored by that cluster. So, this will only connect to a single cluster of Sentinel servers,
+    but will support sharding by asking that sentinel cluster for different services. Also, any Redis connection
+    options (socket timeout, socket keepalive, etc) will be assumed to be identical across Redis server, and
+    across all services.
+
+    "hosts" in this arrangement, is used to list the Redis sentinel hosts. As such, it only supports the
+    tuple method of specifying hosts, as that's all the Redis sentinel python library supports at the moment.
+
+    "services" is the list of Redis services monitored by the sentinel system that Redis keys will be distributed
+     across. If services is empty, this will fetch all services from Sentinel at initialization.
+    """
+
+    def __init__(self, hosts=None, connection_kwargs=None, sentinel_services=None, sentinel_refresh_interval=0):
+        # Master connection caching
+        self._sentinel_refresh_interval = sentinel_refresh_interval
+        self._last_sentinel_refresh = 0
+        self._master_connections = {}
+
+        self._sentinel = redis.sentinel.Sentinel(self._setup_hosts(hosts), **(connection_kwargs or {}))
+        if sentinel_services:
+            self._validate_service_names(sentinel_services)
+            self._services = sentinel_services
+        else:
+            self._services = self._get_service_names()
+        self._ring_size = len(self._services)
+        self._receive_index_generator = itertools.cycle(range(self._ring_size))
```

Comment:
`_receive_index_generator` is never used.

Reply:
Good catch. Fixed.